### ISSUE TITLE
Build fails with `open /dev/null: permission denied`

### ISSUE THREAD

**Issue by cburgmer** [concourse/concourse#6868]
## Summary

Build fails with internal error `open /dev/null: permission denied`

## Steps to reproduce

```sh
# tab 1
curl -O https://concourse-ci.org/docker-compose.yml
docker-compose up
# tab 2
fly login -t test -c 'http://localhost:8080' -u test -p test
curl -LO https://raw.githubusercontent.com/cburgmer/build-facts/master/examples/concourse/anotherpipeline.yml
fly -t test set-pipeline -p somepipeline -c anotherpipeline.yml -n
fly -t test unpause-pipeline -p somepipeline
fly -t test unpause-job -j somepipeline/hello-world
fly -t test trigger-job -j somepipeline/hello-world
```

## Expected results

Job is run succesfully.

## Actual results

Build fails with the following event output:

```
task: say-hello

selected worker: 6733d9790e24

find or create container on worker 6733d9790e24: runc run: exit status 1: container_linux.go:367: starting container process caused: process_linux.go:459: container init caused: rootfs_linux.go:121: reopening /dev/null inside container caused: Failed to open /dev/null - open /dev/null: permission denied
```

## Additional context

```
$ docker-compose --version
docker-compose version 1.29.0, build 07737305
$ which docker-compose
/usr/local/bin/docker-compose
$ ls -al /usr/local/bin/docker-compose
lrwxr-xr-x  1 cburgmer  wheel  77 Feb  8  2020 /usr/local/bin/docker-compose -> /Applications/Docker.app/Contents/Resources/bin/docker-compose/docker-compose
```

## Triaging info

* Concourse version: v7.1.0
* Browser (if applicable): n/a
* Did this used to work? ja

**Comment by aoldershaw** [comment]
I'm seeing the same actually after upgrading Docker Desktop (for Mac) to 3.3.0

**Comment by taylorsilva** [comment]
+1 what Aidan said. I was just on docker 3.2.2 and wasn't getting this error. I then upgraded to 3.3.0 to see if I would get this error and I do.

**Comment by aoldershaw** [comment]
Seems related to https://github.com/docker/for-mac/issues/5559

**Comment by muntac** [comment]
They're tracking the issue here [now](https://github.com/docker/for-mac/issues/5527). And someone [mentioned](https://github.com/docker/for-mac/issues/5527#issuecomment-817284950) it should be fixed in 3.3.1 out next week.
